### PR TITLE
added codecov workflows #306

### LINKED ISSUE #306: Add codecov
Investigate github actions and codecov more generally then implement codecov into this repo.

Ensure it ignores e2e tests. This would be useful to see on PRs as I rarely run the coverage report myself.

See https://github.com/dafnifacility/cli/blob/main/.github/workflows/test.yml
and https://docs.codecov.com/docs/code-coverage-with-python

Will need the codecov token added as a secret by an admin.

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

see #306

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
closes #306

### CODE REVIEW

**Comment by MatteoGuarnaccia5** [comment]
Check ci when reviewing to see that codecov report is uploaded correctly and **only** for unit tests

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  status:
+    project:
+      default:
+        target: auto
+        threshold: 5
+    patch:
+      default:
+        target: 75
+ignore:
+  - "test/"
```
Comment (joelvdavies) [inline codecov.yml:11]: Why does this ignore all tests? Shouldn't it just be the e2e ones?

Comment (MatteoGuarnaccia5) [reply codecov.yml:11]: This is not to ignore the e2e tests, it ignores coverage in the actual tests, ims has a similar thing for mock data, I initially had it to be for just the mock data as well which I can change it back to

Comment (MatteoGuarnaccia5) [reply codecov.yml:11]: https://github.com/ral-facilities/inventory-management-system/blob/develop/codecov.yml

Comment (joelvdavies) [reply codecov.yml:11]: Oh yeah, sorry you're right

Comment (joelvdavies) [reply codecov.yml:11]: Feel free to merge.

**Pull request merged**